[Atharv Joshi] Duke increments

atharvjoshi opened the pull request:
Duke at iP Week 2 deliverable stage

j-lum committed:
Add support for Gradle workflow

j-lum committed:
Add sample checkstyle configuration

Add toolVersion block in to Gradle code sample to prevent errors.

j-lum committed:
Change file mode on `gradle` to be executable

j-lum committed:
Merge pull request #12 from j-lum/gradle+x

Change file mode on `gradle` to be executable (#9)

j-lum committed:
Add configuration for console applications

Gradle defaults to an empty stdin which results in runtime exceptions
when attempting to read from `System.in`. Let's add some sensible
defaults for students who may still need to work with the standard
input stream.

j-lum committed:
Merge pull request #13 from j-lum/javaexec

Add configuration for console applications

atharvjoshi committed:
Created driver class for Duke

atharvjoshi committed:
Encapsulated Duke into separate class

atharvjoshi committed:
Duke now greets with a welcome message

atharvjoshi committed:
Duke now greets and processes commands (echo or exit)

atharvjoshi committed:
Add command storage and retrieval capability to Duke

atharvjoshi committed:
Add task status tracking capability to Duke

Duke can now mark tasks as done, and reflect the status of the task (done or not done) whilst printing the list of tasks. Created a new class 'Task' for better encapsulation. Text encoding for 'Task Status Icon' fishy in my machine (IDE prints question marks instead of unicode characters), but the logic of the code is present.

atharvjoshi committed:
Add capability to handle various types of events.

These events includes todos, deadlines, and events.

atharvjoshi committed:
Enable Duke to handle errors concerning badly formatted commands.

atharvjoshi committed:
Add support for deleting tasks and handling more errroneous input cases.

At level 6, Duke is now able to manage three different types of tasks - todos, deadlines, and events. Duke can add, delete and change the 'done' status of these tasks. She can also inform the user when they have entered an incorrect command, input in a wrong format, or an unreasonable query. While Duke can perform these basic functions now, there remains scpe for improvement. For instance, instead of using switch statements to toggle between handling of different commands, perhaps enums might offer a more efficient way. Furthermore, all testing done on Duke has been entirely manual - I/O redirection to perform semi-automated testing might be another logical direction to extend this project in.

atharvjoshi committed:
Add more thorough exception handling, enums for enforcing type safety.

atharvjoshi committed:
Add functionality to save task list to and load from hard disk.

atharvjoshi committed:
Add ability to understand dates and times.

atharvjoshi committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Add functionality to save task list to and load from hard disk.

atharvjoshi committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Add ability to understand dates and times.

# Conflicts:
#	src/main/java/Deadline.class
#	src/main/java/Duke$Command$1.class
#	src/main/java/Duke$Command$2.class
#	src/main/java/Duke$Command$3.class
#	src/main/java/Duke$Command$4.class
#	src/main/java/Duke$Command$5.class
#	src/main/java/Duke$Command$6.class
#	src/main/java/Duke$Command$7.class
#	src/main/java/Duke$Command.class
#	src/main/java/Duke.class
#	src/main/java/Duke.java
#	src/main/java/Event.class

atharvjoshi committed:
Make code more object-oriented via handler classes

atharvjoshi committed:
Add preliminary JUnit test methods.

atharvjoshi committed:
Add javadoc comments.

atharvjoshi committed:
Comply with Java coding standard.

atharvjoshi committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Add javadoc comments.

atharvjoshi committed:
Merge branch 'branch-A-CodingStandard'

atharvjoshi committed:
Comply with Java Coding Standard.

dragontho reviewed (commented):
Nothing to say, if anything I need to ask you for help to make my code look nicer HAHA. All the principles of software engineering is there (at least those i am aware of).

atharvjoshi committed:
Organise project into packages.

atharvjoshi committed:
Add working JUnit Tests for ToDo and TaskList classes.

atharvjoshi committed:
Merge branch 'gradle' of https://github.com/atharvjoshi/duke

atharvjoshi committed:
Automate build projects using Gradle.

atharvjoshi committed:
Package the app as a JAR file.

atharvjoshi committed:
Give users a way to find a task by searching for a keyword.

atharvjoshi committed:
Merge branch 'branch-Level-9'

atharvjoshi committed:
Use checkStyle to detect coding style violations.

atharvjoshi committed:
Merge branch 'branch-A-Checkstyle'

atharvjoshi committed:
Add a GUI to Duke using JavaFX.

atharvjoshi committed:
Merge branch 'branch-Level-10'

atharvjoshi committed:
Use assertions to ensure code logic.

atharvjoshi committed:
Merge pull request #4 from atharvjoshi/branch-A-Assertions

Atharv Joshi - Duke Increments

atharvjoshi committed:
Merge branch 'master' of https://github.com/atharvjoshi/duke

atharvjoshi committed:
Examine the code and refactor to improve the code quality where necessary.

atharvjoshi committed:
Merge pull request #5 from atharvjoshi/branch-A-CodeQuality

Atharv Joshi - Duke Increments 3

atharvjoshi committed:
Merge branch 'master' of https://github.com/atharvjoshi/duke

atharvjoshi committed:
Provide a way to undo the last add, delete, and done command.

Only works for current Duke session i.e. tasks loaded from storage cannot be undone when a new Duke session is opened.

atharvjoshi committed:
Merge branch 'branch-C-Undo'

shawnkoh commented on src/main/java/commands/DoneCommand.java:
```diff
@@ -0,0 +1,102 @@
+                } else {
+                    return ui.showTaskAlreadyDoneMessage();
+                }
+            } catch (IndexOutOfBoundsException exceptionOne) {
```
avoid catching runtime exceptions - they should be handled using if/else instead. a good example of this would be user input errors (e.g. delete a instead of delete 2)
exceptions should be for truly exceptional cases that are out of your control; such as when you try to save the data to the file and for some reason it does not work i.e. IOException. For exceptions that you can control like FileNotFoundException, they should be handled.

shawnkoh commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,197 @@
+     * Enumeration of all command types supported by Duke.
+     */
+    private enum Imperative {
+        BYE, DEADLINE, DELETE, DONE, EVENT, FIND, LIST, TODO, UNDO
```
consider declaring a value for your enums i.e. BYE ("bye"), DEADLINE ("deadline")
This prevents the scenario where a user can provide 0, 1, 2 - currently your enums are defaulted to 0, 1 and 2.

shawnkoh commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,225 @@
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        this.welcomeMessage = "Hello! I'm Duke!!!\n";
```
consider declaring welcomeMessage as a constant i.e. private static final

shawnkoh commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,225 @@
+     * @return startup error message.
+     */
+    String showStartupError() {
+        return "Unable to load tasks from hard disk\n";
```
inconsistent return - one is a variable and the other a string.
Consider using System.lineSeperator() instead to support all platforms instead of \n

atharvjoshi committed:
Create .travis.yml

atharvjoshi committed:
Update README.md

Add a User Guide

atharvjoshi committed:
Update README.md

atharvjoshi committed:
Add a User Guide

atharvjoshi committed:
Add a product screenshot.

atharvjoshi committed:
Update the User Guide

atharvjoshi committed:
Relase the product.

atharvjoshi committed:
Make the jar executable cross-platform.

atharvjoshi committed:
Update README.md

atharvjoshi committed:
Add product screenshot

atharvjoshi committed:
Merge branch 'master' of https://github.com/atharvjoshi/duke

atharvjoshi committed:
Add product screenshot with proper name format

atharvjoshi committed:
Add product screenshot attempt #3

atharvjoshi committed:
Add product screenshot attempt #4

atharvjoshi committed:
Add product screenshot successfully.

atharvjoshi committed:
Update README.md

atharvjoshi committed:
Merge branch 'master' of https://github.com/atharvjoshi/duke

atharvjoshi committed:
Fix bug so that program creates a directory and file to save task list to user's disk.

atharvjoshi committed:
Update README.md

atharvjoshi committed:
Remove product screenshot to be added again.

atharvjoshi committed:
Merge branch 'master' of https://github.com/atharvjoshi/duke

atharvjoshi committed:
Add product screenshot (attempt #7)

atharvjoshi committed:
Update README.md

atharvjoshi committed:
Update README.md

atharvjoshi committed:
Merge branch 'master' of https://github.com/atharvjoshi/duke
